File: Data Structures & Algorithms/max-water-container/submission-5.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left=0
        right=len(heights)-1
        best=0
        
        while left<right:
            h = min(heights[left], heights[right])
            width = right - left
            currentArea = h*width
            best = max(currentArea, best)

            if(heights[left]<heights[right]):
                left+=1
            else:
                right-=1
        return best

        

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-5.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashSet<Character> window = new HashSet<>();
        int left=0;
        int best = 0;
        for(int right=0 ; right<s.length(); right++){
             Character ch = s.charAt(right);
             while(window.contains(ch)){
                window.remove(s.charAt(left));
                left++;
             }
             window.add(ch);
             best = Math.max(best, right-left+1);
        }
        return best;
    }
}
